command/pr_review.go

```diff
@@ -128,10 +134,128 @@ func prReview(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if input == nil {
+		input, err = reviewSurvey(cmd)
+		if err != nil {
+			return err
+		}
+		if input == nil && err == nil {
+			// Cancelled.
+			return nil
+		}
+	}
+
 	err = api.AddReview(apiClient, pr, input)
 	if err != nil {
 		return fmt.Errorf("failed to create review: %w", err)
 	}
 
 	return nil
 }
+
+func reviewSurvey(cmd *cobra.Command) (*api.PullRequestReviewInput, error) {
+	editorCommand, err := determineEditor(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	typeAnswers := struct {
+		ReviewType string
+	}{}
+	typeQs := []*survey.Question{
+		{
+			Name: "reviewType",
+			Prompt: &survey.Select{
+				Message: "What kind of review do you want to create?",
+				Options: []string{
+					"Comment",
+					"Approve",
+					"Request Changes",
+					"Cancel",
+				},
+			},
+		},
+	}
+
+	err = SurveyAsk(typeQs, &typeAnswers)
+	if err != nil {
+		return nil, err
+	}
+
+	reviewState := api.ReviewComment
+
+	switch typeAnswers.ReviewType {
+	case "Approve":
+		reviewState = api.ReviewApprove
+	case "Request Changes":
+		reviewState = api.ReviewRequestChanges
+	case "Cancel":
+		return nil, nil
+	}
+
+	bodyAnswers := struct {
+		Body string
+	}{}
+
+	bodyQs := []*survey.Question{
+		&survey.Question{
+			Name: "body",
+			Prompt: &surveyext.GhEditor{
+				EditorCommand: editorCommand,
+				Editor: &survey.Editor{
+					Message:  "Review Body",
+					FileName: "*.md",
+				},
+			},
+		},
+	}
+
+	err = SurveyAsk(bodyQs, &bodyAnswers)
+	if err != nil {
+		return nil, err
+	}
+
+	if reviewState == api.ReviewComment && bodyAnswers.Body == "" {
+		return nil, errors.New("cannot leave blank comment")
+	}
+
+	if len(bodyAnswers.Body) > 0 {
+		out := colorableOut(cmd)
+		renderedBody, err := utils.RenderMarkdown(bodyAnswers.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		fmt.Fprintf(out, "Got:\n%s", renderedBody)
+	}
+
+	confirmAnswers := struct {
+		Confirm string
+	}{}
+	confirmQs := []*survey.Question{
+		{
+			Name: "confirm",
+			Prompt: &survey.Select{
+				Message: "What's next?",
+				Options: []string{
+					"Submit",
+					"Cancel",
```

Comment:
As I understand, right now this flow has 3 steps:
1. “What kind of review do you want to create?”
  - Comment
  - Approve
  - Request changes
  - Cancel
2. “Review body” (opens text editor)
3. “What's next?”
  - Submit
  - Cancel

Have we considered trying to get away with just 2 prompts?

1. “Review body” (opens text editor)
2. “What's next?”
  - Comment
  - Approve
  - Request changes
  - Cancel

I think this would be both simpler for our users and require less code. /cc @ampinsk

Reply:
Thanks for the suggestion @mislav! I find conflating the submit step with choosing the type of review to be a little confusing. I think giving them their own prompts while slightly more steps, makes more semantic sense. 

I was also thinking for this we could do just a simpler `Submit? [Y/n]` instead of `What's next?`

Reply:
I switched to the Y/n prompt